src/PactNet.Native/MessageScenarioBuilder.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+
+namespace PactNet.Native
+{
+    /// <summary>
+    /// Defines the message scenario builder
+    /// </summary>
+    public class MessageScenarioBuilder : IMessageScenarioBuilder
+    {
+        private MessageScenarioBuilder()
+        {
+        }
+
+        /// <summary>
+        /// Handles to create a new scenario with scenario builder
+        /// </summary>
+        public static IMessageScenarioBuilder AScenario => new MessageScenarioBuilder();
```

Comment:
This feels quite clunky - can't the `WhenReceiving` method below be used instead?

Reply:
No that won't do. The only way around would be to put public the constructor and create an instance of the builder. The syntax is just horrible to write.
In the consumer approach, you are using an extensions method to do the heavy lifting for you and since it has different version, it allows more flexibility.

Therefore we can either:
1) use another set of extensions method for scenario setting (maybe by specifying a pact version) 
2) or a static handle, like here, which is a common practice in fluent builders.